src/main/java/frc/robot/subsystems/intake/IntakeSubsystem.java

```diff
@@ -299,42 +383,99 @@ public void simulationPeriodic() {
     pivotArmMech.setAngle(Radians.of(pivotSim.getAngleRads()).in(Degrees));
 
     switch (state) {
-      case Idle:
-        if (MathUtil.isNear(
-                pivotSim.getAngleRads(), kPivotExtendAngle.in(Radians), kPivotTolerance.in(Radians))
-            && Math.abs(intakeMotor.get()) > 0
-            && Math.abs(conveyorMotor.get()) > 0) {
-          timer.start();
+      case IDLE:
+        sensors.setSimState(CoralEnum.NO_CORAL);
+
+        if (atSetpoint(kPivotExtendAngle) && intakeMotor.get() > 0) {
+          t.start();
         } else {
-          timer.stop();
+          t.stop();
         }
 
-        if (timer.hasElapsed(2)) {
-          simbeam.set(true);
-          state = IntakeState.atSensor;
-          timer.reset();
+        if (t.hasElapsed(0.5)) {
+          state = IntakeState.FLOOR_INTAKE;
         }
-        break;
-      case atSensor:
-        if (Math.abs(conveyorMotor.get()) > 0) {
-          if (!timer.isRunning()) {
-            timer.start();
-          }
+
+        if (atSetpoint(kcoralStation) && intakeMotor.get() > 0) {
+          t.start();
```

Comment:
You'll need seperate Timers if you want to be able to track progress to different states like this.